src/Datadog.Trace/Util/Http/HttpRequestExtensions.Core.cs

```diff
@@ -19,9 +19,9 @@ internal static partial class HttpRequestExtensions
         internal static Dictionary<string, object> PrepareArgsForWaf(this HttpRequest request, RouteData routeDatas = null)
         {
             var url = GetUrl(request);
-
             var headersDic = new Dictionary<string, string>();
-            foreach (var k in request.Headers.Keys)
+            var headerKeys = request.Headers.Keys.Where(k => k != "Cookie");
```

Comment:
Don't know if you want to use LINQ here, and instead move to be an `if` inside the loop (as you do in the other implementation for FX?). Also, probably want to make the comparison case insensitive?

Reply:
thanks, done in 4b5c2315584c64409bd566bde67b81b8489e2807